### PR TITLE
Added basic line indenting to HCL languages.

### PR DESCRIPTION
**Pull request by lkishalmi**
Making HCL editing a bit easier...

### CODE REVIEW

File: ide/languages.hcl/src/org/netbeans/modules/languages/hcl/BasicHCLLexer.java [hunk 1]
```diff
@@ -30,93 +31,59 @@
...
+        } else {
+            return token(id);
         }
     }
+/*
+    @Override
+    protected String flyweightText(HCLTokenId id) {
+        return id.getFixedText();
+    }
+*/
```
Comment (matthiasblaesing) [inline ide/languages.hcl/src/org/netbeans/modules/languages/hcl/BasicHCLLexer.java:81]: Intentionally commented out?

**Review by matthiasblaesing — COMMENTED** [review]
Looks mostly sane to me. I have one problem with this: It introduces trailing whitespace on empty lines. I noticed this when I tested this with the `example.tf` file. It looked (and is) well indented, yet the file becomes modified, because the empty lines in the blocks get space injected.

**Comment by lkishalmi** [comment]
Thanks for reporting that!

I'm working on some tests, as some of the (edge) cases might be not handled well. And now I wish we have Java 17 Textblocks...

**Review by matthiasblaesing — APPROVED** [review]
Eyeballed this and looks sane to me. My tests, though limited, worked ok.

**Review by vieiro — APPROVED** [review]
Go for it, @lkishalmi !

**Pull request merged**